Master

inetuid opened the pull request:
Please review for initial merge.

dbarrosop commented:
There is some stuff missing. I would recommend you taking a look to the skeleton:

https://github.com/napalm-automation/napalm-skeleton

For example:

1. The mocked tests are missing. Let me know if you need help with this. You can check this as an example: https://github.com/napalm-automation/napalm-eos/blob/develop/test/unit/TestEOSDriver.py#L60
1. Methods that are not part of the base definition add an underscores as prefix. For example, `get_mndp_neighbors` -> `_ get_mndp_neighbors`.
1. The documentation updates in `napalm` are missing.
1. The changes on `napalm-base` are missing as well.

I would focus first on fixing the mocked tests. Then I can start helping you integrating the project with travis and publishing the driver on pip before updating `napalm` and `napalm-base`. If you are unsure how to do that

dbarrosop commented:
Two more things:

1. Can you send the PR against the `develop` branch?
2. Looks like you are using tabs for indenting the text. Could you use 4 spaces instead, please?

inetuid commented:
Progress thus far:

1. I have added the mock tests for TestGetterROSDriver.
2. Not sure what I need to do for the configuration tests. Is it a Vagrant box you need? Any specific usernames/passwords, IP addresses, routing protocols etc?
3. Renamed methods that are not part of base definition.
4. TBD
5. TBD

Two more things:
1. PR has been sent to develop.
2. Tabs have been replaced by 4 spaces.

Let me know on the configuration questions and I'll start work on them.


Matt.
